examples/04_fuzzy_joining.py

```diff
@@ -416,7 +419,13 @@
 from sklearn.model_selection import GridSearchCV
 
 # We will test four possible values of match_score:
-params = {"joiner__match_score": [0.2, 0.3, 0.4, 0.5]}
```

Comment:
here AFAICT all the considered hyperparameters result in the same transformation (`match_score` is low and no fuzzy match is rejected). The selected hyperparameter is the first we specify in the list because they all get the same score. If we increase the threshold so that some rows don't match, we get some [`pd.NA`](https://github.com/skrub-data/skrub/blob/a8af7d5869d7a8084c46b013efb4d6a1bd88182c/skrub/_fuzzy_join.py#L513) in the transform output and the gradient boosting raises an exception.

on the `main` branch, the same problem exists, but when we increase the `match_score`,  the join crashes with a different error before we get to the gradient boosting, because the `Joiner` uses the same suffix for all tables, which results in duplicate column names in this example, which causes a cryptic pandas error when trying to assign values to a part of the dataframe [here](https://github.com/skrub-data/skrub/blob/a8af7d5869d7a8084c46b013efb4d6a1bd88182c/skrub/_fuzzy_join.py#L512)